.travis.yml

```diff
@@ -55,7 +57,20 @@ jobs:
         - git config --global user.email "${GH_EMAIL}"
         - echo "machine github.com login ${GH_NAME} password ${GH_TOKEN}" > ~/.netrc
         - cd website && yarn install && GIT_USER="${GH_NAME}" yarn run publish-gh-pages
-        - git checkout gh-pages && git checkout $(git rev-list -n1 HEAD -- index.yaml)~1 -- index.yaml && git add index.yaml && git commit -m "Auto-restore Helm chart repo index after Docusaurus publish" && git push --verbose
+    - stage: restore-helm-repo-index
+      language: minimal
+      script:
+        - git config --global user.name "${GH_NAME}"
+        - git config --global user.email "${GH_EMAIL}"
+        - echo "machine github.com login ${GH_NAME} password ${GH_TOKEN}" > ~/.netrc
+        - git fetch --unshallow
+        - git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
+        - git fetch origin
+        - git checkout gh-pages
+        - git checkout $(git rev-list -n1 HEAD -- index.yaml)~1 -- index.yaml > /dev/null 2>&1 | true
```

Comment:
why ignore stderr here?

Reply:
as I wanted this stage to also work if the HEAD already has index.yaml and `$(git rev-list -n1 HEAD -- index.yaml)~1` doesn't. It won't happen actually, but it happened during testing. Hmm, actually this was the explanation of `| true`. We could remove ignoring stderr, but then we could see an error but the pipeline doesn't fail.

Reply:
ah sure, the file might exist... perhaps a more explicit way of handling this case would be:

```bash
test -e index.yaml || git checkout $(git rev-list -n1 HEAD -- index.yaml)~1 -- index.yaml
```

Reply:
I like that... will change it.